apis/core.oam.dev/v1alpha2/core_types.go

```diff
@@ -24,6 +24,20 @@ import (
 	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
 )
 
+// CUE defines the encapsulation in CUE format
+type CUE struct {
+	// Template defines the abstraction template data of the capability, it will replace the old CUE template in extension field.
+	// Template is a required field if CUE is defined in Capability Definition.
+	Template string `json:"template"`
+}
+
+// Schematic defines the encapsulation of this capability(workload/trait/scope),
+// the encapsulation can be defined in different ways, e.g. CUE/HCL(terraform)/KUBE(K8s Object)/HELM, etc...
+type Schematic struct {
+	CUE *CUE `json:"cue,omitempty"`
```

Comment:
There are some discussions about using a plug-in model instead of hardcode each schematic

Reply:
The agreement is we keep it as is until another new schematic type besides CUE, Helm, Kube, ARM comes in.